### PR TITLE
Added Exists detector (merge this after date range)

### PR DESCRIPTION
**Pull request by chollinger**
# The Exists Detector
###### USE THIS TEMPLATE FOR YOUR PULL REQUEST!

  * Primary Contact: [Cory Hollinger](mailto:cory.hollinger@gmail.com)

### Purpose

Added a simple boolean Exists detector.

### Expected Outcome

Configs no longer have to use regex to detect existence 
  
### Checklist
 - [x] Local test from submitter
 - [x] Maintainer Code Review
 - [ ] Added To Wiki

### CODE REVIEW

**Comment by chollinger** [comment]
Do you want test logs for this, or are the unit tests enough? I marked it WIP just in case, but it's ready to go if unit tests are enough.

File: tgt_grease_enterprise/Detectors/Exists.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    def __init__(self):
+        self._result = {'result': False}
+
+    def param_compute(self, source, rules):
+        rules_passed = 0
+        rules_needed_to_pass = len(rules)
+
+        for rule in rules:
+            if 'field' not in rule:
+                self._result['result'] = False
```
Comment (lemoney) [inline tgt_grease_enterprise/Detectors/Exists.py]: Can we get some inline comments on this code

File: tgt_grease_enterprise/Detectors/Exists.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+from .BaseDetectorClass import BaseDetector
+
+
```
Comment (lemoney) [inline tgt_grease_enterprise/Detectors/Exists.py]: Can you place a doc block here for what the logical block should look like

File: tgt_grease_enterprise/Detectors/tests/test_exists.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            {
+                'biscuits': 'gravy'
+            }
+        ]
+
+        self.Exists.param_compute(source=source_data, rules=rules)
+        self.assertFalse(self.Exists.get_result()['result'])
+        self._teardown()
+        
+    def test_variable_set(self):
```
Comment (lemoney) [inline tgt_grease_enterprise/Detectors/tests/test_exists.py]: This is high quality testing! good work!

**Comment by chollinger** [comment]
@lemoney @gamagori This one's been updated too

**Comment by chollinger** [comment]
For adding this to the wiki, where is the best place to fit it in?

File: tgt_grease_enterprise/Detectors/tests/test_exists.py (hunk 3 above)
Comment (chollinger) [reply tgt_grease_enterprise/Detectors/tests/test_exists.py]: Gee, thanks!

**Pull request merged**